core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -741,18 +742,18 @@ class DAGScheduler(
       val stageInfos = stageIds.flatMap(id => stageIdToStage.get(id).map(_.latestInfo))
       listenerBus.post(
         SparkListenerJobStart(job.jobId, jobSubmissionTime, stageInfos, properties))
-      submitStage(finalStage)
+      submitStage(finalStage, Some(missingStages))
     }
     submitWaitingStages()
   }
 
   /** Submits stage, but first recursively submits any missing parents. */
-  private def submitStage(stage: Stage) {
+  private def submitStage(stage: Stage, missingStages: Option[List[Stage]] = None) {
```

Comment:
There are a number of recursive calls that then don't get this value then and recompute anyway? To avoid complication, is it simpler to just have all callers provide this info? I don't think it's but one more change.